draft-ietf-privacypass-batched-tokens.md

```diff
@@ -437,11 +480,11 @@ client. The issuer creates a BatchTokenResponse structured as follows:
 
 ~~~tls
 struct {
-  TokenResponse token_response<0..2^16-1>; /* Defined by token_type */
+  optional<TokenResponse> token_response; /* Defined by token_type */
 } OptionalTokenResponse;
 
 struct {
-  OptionalTokenResponse token_responses<0..2^16-1>;
+  OptionalTokenResponse token_responses<V>;
```

Comment:
If `token_responses[i]` is empty, it is just encoded as `0x00`.
If it is non-empty, it is encoded as `0x01` followed by its regular encoding.

In other words, step 1 from above is not needed.

This is also what MLS uses and therefore some implementations already support this optional encoding.

Reply:
what I'm referring to is each element of `token_responses` array is prefixed by its length. Therefore, even if the empty option is `0x00`, it's an element of length `0x01`.

I think what we want is an array where the length prefix denotes the presence of the object. A length of `0x00` means the option is `None`. A length greater than 0 means the option is `Some`.

Reply:
> what I'm referring to is each element of token_responses array is prefixed by its length.

It's the other way around, each element is now prefixed by the option value (`0x00` or `0x01`) and only then prefixed by the length of the array (in case the option value is `0x01`).

Reply:
taking a new look at [RFC5246](https://datatracker.ietf.org/doc/html/rfc5246) again, this was an issue from my side. I thought TLS vectors always prefixed elements of the sequence with their length. The RFC defines the opposite: vectors are not prefixed with their length.

Reply:
Ok to resolve. I don't have these permissions on the repository